### ISSUE TITLE
NPM install timeout on macOS-13

### ISSUE THREAD

**Issue by perpifran** [actions/runner-images#10720]
### Description

When doing an `npm install` in the execution of an Azure Devops pipeline, the execution of `npm install` takes 1 hour and the machine timeout is triggered. Previously this step took 5-7 minutes, and with the new version this happens. I haven't modified anything about the pipeline or the project in months. The Azure Pipeline log does not provide any more useful information.

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [X] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] macOS 15
- [ ] macOS 15 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

20240916.98

### Is it regression?

20240616.1

### Expected behavior

`npm install` execution should take a reasonably short time

### Actual behavior

The `npm install` execution takes more than 1 hour and the machine stops it by launching the timeout

### Repro steps

1. Use a macos-13 image
2. Install NodeJs version 18.x
3. Try to execute `npm install` on ionic repo

**Comment by erik-bershel** [comment]
Hey @perpifran!

Please provide more detailed repro steps. I tested next with positive result:
```yml
jobs:
  test:
    strategy:
      matrix:
        os: [ macos-12, macos-13, macos-14, macos-15 ]
    runs-on: ${{ matrix.os }}

    steps:
      - name: Checkout Ionic Framework repository
        run: |
          git clone https://github.com/ionic-team/ionic-framework.git

      - name: Set up Node.js 18.x
        uses: actions/setup-node@v3
        with:
          node-version: '18.x'

      - name: Show Node and NPM versions
        run: |
          node -v
          npm -v

      - name: Install project dependencies
        working-directory: ionic-framework
        run: |
          npm install
        timeout-minutes: 60
```

Result:

<img width="1073" alt="Screenshot 2024-10-02 at 11 38 29" src="https://github.com/user-attachments/assets/66228029-8364-4273-b67a-9f030503c8fc">

**Comment by perpifran** [comment]
Hi @erik-bershel thanks for your time.

Here I give you an example of a pipeline executed in Azure Devops that gives me a timeout error.
```yml
trigger:
  - none

variables:
  - name: nodeVersion 
    value: '18.x'

stages:
  - stage: Build_iOS
    displayName: Build iOS App
    jobs:
      - job:
        displayName: Build iOS App
        pool:
          vmImage: 'macos-13'
        steps:
          - task: NodeTool@0
            inputs:
              versionSpec: '$(nodeVersion)'
            displayName: 'Install Node.js version $(nodeVersion)'

          - task: Npm@1
            displayName: Npm Install
            inputs:
              command: 'custom'
              customCommand: 'ci'
              workingDir: '$(Build.SourcesDirectory)'
```

I also attach the log of the step where I execute the installation of dependencies.
[4_Npm Install.txt](https://github.com/user-attachments/files/17246851/4_Npm.Install.txt)

I hope it is helpful. Thanks.

**Comment by perpifran** [comment]
Hi, is a lot of dependencies, and we have our own private packages called `*cnts*`. If necessary I can do a test with something smaller.

[package.json](https://github.com/user-attachments/files/17255340/package.json)
[package-lock.json](https://github.com/user-attachments/files/17255342/package-lock.json)

**Comment by erik-bershel** [comment]
Tried next on ADO to check functionality:

```yml
trigger: none

stages:
  - stage: ManualPipeline
    displayName: 'Manual Pipeline'
    jobs:
      - job: Test
        displayName: 'Test on MacOS'
        strategy:
          matrix:
            12:
              imageName: 'macos-12'
            13:
              imageName: 'macos-13'
            14:
              imageName: 'macos-14'

        pool:
          vmImage: $(imageName)
        steps:
          - checkout: none
          - script: |
              git clone https://github.com/ionic-team/ionic-framework.git
            displayName: 'Checkout Ionic Framework repository'

          - task: UseNode@1
            inputs:
              version: '18.x'
            displayName: 'Set up Node.js 18.x'

          - script: |
              node -v
              npm -v
            displayName: 'Show Node and NPM versions'

          - script: |
              npm install
            workingDirectory: 'ionic-framework'
            displayName: 'Install project dependencies'
            timeoutInMinutes: 60
```

macOS-13 is slower than the others, but not so drastically:

<img width="1149" alt="Screenshot 2024-10-04 at 13 53 38" src="https://github.com/user-attachments/assets/d4e214d3-aee6-4229-b561-3acdd0e2731a">

Install dependencies step is 30-40% slower then on other macOS releases.

**Comment by erik-bershel** [comment]
And I tried this:
```yml
trigger: none

stages:
  - stage: ManualPipeline
    displayName: 'Manual Pipeline'
    jobs:
      - job: Test
        displayName: 'Test on MacOS'
        strategy:
          matrix:
            12:
              imageName: 'macos-12'
            13:
              imageName: 'macos-13'
            14:
              imageName: 'macos-14'

        pool:
          vmImage: $(imageName)
        steps:
          - checkout: none
          - script: |
              git clone https://github.com/ionic-team/ionic-framework.git
            displayName: 'Checkout Ionic Framework repository'

          - task: UseNode@1
            inputs:
              version: '18.x'
            displayName: 'Set up Node.js 18.x'

          - script: |
              node -v
              npm -v
            displayName: 'Show Node and NPM versions'

          - task: Npm@1
            displayName: Npm Install
            inputs:
              command: 'custom'
              customCommand: 'ci'
              workingDir: 'ionic-framework'
            timeoutInMinutes: 60
```

Results are still positive:

<img width="1152" alt="Screenshot 2024-10-04 at 14 18 24" src="https://github.com/user-attachments/assets/8bc955b0-74ec-4826-8496-a5d9ee1e0c76">

Install dependencies step with `npm ci` command shows about 10-20% difference across macOS releases.

**Comment by erik-bershel** [comment]
> Hi, is a lot of dependencies, and we have our own private packages called *cnts*. If necessary I can do a test with something smaller.

It seems that the issue hiding exactly in one of your dependencies but not in the image itself since it is not reproducible on the example project. Unfortunately we do not maintain custom builds and npm packages, so I forced to close this issue as external. Also I should note that I tried to run repro with your packages list but had no luck since there is private package and it requires authentification.
